src/base/opt/exterr.rs

```diff
@@ -117,15 +120,15 @@ impl<Octs> ExtendedError<Octs> {
     /// correctly encoded UTF-8, returns `Some(Ok(_))`. If there is text but
     /// it is not UTF-8, returns `Some(Err(_))`.
     pub fn text(&self) -> Option<Result<&Str<Octs>, &Octs>> {
-        self.text.as_ref().map(Result::as_ref)
+        self.text.as_ref().map(|res| res.as_ref().map_err(|err| err.as_ref()))
     }
 
     /// Returns the text as an octets slice.
     pub fn text_slice(&self) -> Option<&[u8]>
     where Octs: AsRef<[u8]> {
         match self.text {
             Some(Ok(ref text)) => Some(text.as_slice()),
-            Some(Err(ref text)) => Some(text.as_ref()),
+            Some(Err(ref text)) => Some(text.as_ref().as_ref()),
```

Comment:
It's not required but adding `LossyOctets::as_slice()` would make this more readable ;-)

Reply:
And drop the `AsRef<_>` impl instead?